### DISCUSSION TITLE
Question regarding the services working together

### DISCUSSION THREAD

**Discussion by kiguigui** [immich-app/immich#184] [Q&A] [answered]
Hello there,

i got immich running and its great! :)  I've got a few questions regarding the Immich stack:

- The ENV File UPLOAD_LOCATION is mandatory at least for immich-server. But since im using docker named volumes, it shouldn't be. What is immich-server doing with this path?
- What is the purpose of NGINX in this stack? It looks like its only for the Apps?
- What exactly is needed to put in VITE_SERVER_ENDPOINT? I've struggled hard to understand this but yet im not really sure, whats the purpose of this.

**Comment by alextran1502** [comment, marked as answer]
Hello,

- The UPLOAD_LOCATION is mandatory as it is a commonly used method to mount a network drive or to specify the drive on the file system where they can perform additional actions work, like duplicate the backup to another cloud storage. It is more convenient than having to look into the Docker managed volumes.

- Nginx is for load balancing purpose for the scenario you want to scale the number of the server container

- VITE_SERVER_ENDPOINT is for the web interface to know which endpoint it will need to make the request to. Imagine this is Google Photos, the website is hardcoded to know it needs to make the request back to Google owned service, but since I will have no knowledge of the IP of your server, it will need to be specified by you. So, it is the same server URL you put in the mobile app when logging in

**Comment by kiguigui** [comment]
Hey Alex,

thank you for your detailed response! That has helped my understandig of the services working together :)
